cli_tools/gce_windows_upgrade/upgrader/upgrader.go

```diff
@@ -103,11 +109,22 @@ type upgrader struct {
 	rebootFn                  func() (*daisy.Workflow, error)
 	cleanupFn                 func() (*daisy.Workflow, error)
 	rollbackFn                func() (*daisy.Workflow, error)
+
+	logger logging.Logger
 }
 
 // Run runs upgrader.
 func Run(p *InputParams) (service.Loggable, error) {
-	u := upgrader{InputParams: p}
+	u := upgrader{
+		InputParams: p,
+		logger:      logging.NewToolLogger(logPrefix),
+		derivedVars: &derivedVars{
+			executionID: os.Getenv(daisyutils.BuildIDOSEnvVarName),
+		},
+	}
+	if u.executionID == "" {
+		u.executionID = path.RandString(5)
```

Comment:
I wonder why it doesn't have this exec id and why now we need it

Reply:
The executionID is used when labeling resources (eg: temporary instances and disks):

https://github.com/GoogleCloudPlatform/compute-image-tools/blob/master/cli_tools/common/utils/daisyutils/daisy_worker.go#L60

When running on cloud build, it will use the BUILD_ID environment variable. If it's running outside cloud build, that environment variable may be empty, so it generates a new one.